sdk/identity/identity/src/credentials/defaultAzureCredential.ts

```diff
@@ -27,6 +30,10 @@ export interface DefaultAzureCredentialOptions extends TokenCredentialOptions {
    * This client ID can also be passed through to the {@link ManagedIdentityCredential} through the environment variable: AZURE_CLIENT_ID.
    */
   managedIdentityClientId?: string;
+  /**
+   * Optionally pass in a resource ID to be used by the {@link ManagedIdentityCredential}.
```

Comment:
What does this setting affect (what's the difference between providing it and not providing it), and how does a customer know when/if they should or must provide it?

Reply:
I’m not sure what to put here. I’m investigating.

I see this in the docs:
> (Optional) The Azure resource ID of the user-assigned identity to be used. Cannot be used on a request that includes principal_id, client_id, or object_id. If all ID parameters (client_id, principal_id, object_id, and mi_res_id) are omitted, the system-assigned identity is used.

Source: https://docs.microsoft.com/en-us/azure/app-service/overview-managed-identity?tabs=portal%2Chttp

Reply:
The main point is that this is another way to send the resource information. The main way is to derive it from the scope, however some customers need to have more control of this parameter. See this issue: https://github.com/Azure/azure-sdk-for-net/issues/20651

So I will come up with a better comment here. Thanks!

Reply:
I’ll push this for now:

> Resource IDs may be built by convention, in which cases, this parameter allows users the possibility to fully control what resource ID to use.

Feedback appreciated!

Reply:
So this is mutually exclusive with `managedIdentityClientId`?

The way I look at it is, if someone came and opened an issue on the repo tomorrow and said "hey I see this new field and I'm using Managed Identity in my app, so how do I know if I should be setting this field?" We ought to be able to answer that question and the doc string should clue the user in to that answer.

Saying they may be built by convention doesn't really explain what exactly the convention is and how I or a user would know if the convention does or does not apply to a given use case.

Reply:
If it's the kind of thing where the answer is "you need to use this if you know you need to use it," then I honestly think that's fine, but it'd be good to put something in the doc string that indicates it's a niche use case that likely doesn't apply and a link to MSAL docs that describe precisely the effect of setting this property.

Reply:
It’s not really an MSAL thing, it’s just a protocol thing. I’ll ask around, I’ll ask around! Thank you for pointing me towards this.

Reply:
There are scenarios, such as when user assigned identities are created using an ARM template, where the resource id of the identity is known, but the client id can't be known ahead of time. This allows programs to use these user assigned identities without having to first determine the client id of the created identity.

Reply:
I’ll quote something like that, @schaabs 🙂 thank you!